source/ast/expressions/AssertionExpr.cpp

```diff
@@ -417,6 +490,71 @@ void SequenceRange::serializeTo(ASTSerializer& serializer) const {
         serializer.write("max", "$"sv);
 }
 
+bool SequenceRange::operator!=(const SequenceRange& right) const {
+    if (!max.has_value() && !right.max.has_value())
+        return min != right.min;
+
+    if (max.has_value() && right.max.has_value())
+        return (min != right.min) && (max.value() != right.max.value());
+
+    return false;
+}
+
+bool SequenceRange::operator==(const SequenceRange& right) const {
+    return !(*this != right);
+}
+
+bool SequenceRange::operator>(const SequenceRange& right) const {
+    if (!max.has_value() && !right.max.has_value())
+        return min > right.min;
+
+    if (max.has_value() && right.max.has_value()) {
+        uint32_t maxVal = max.value();
+        uint32_t rightMaxVal = right.max.value();
+        return (min >= right.min) && ((maxVal - rightMaxVal) > 0);
+    }
+
+    return false;
+}
+
+bool SequenceRange::operator<(const SequenceRange& right) const {
+    return !(*this > right) && (*this != right);
+}
+
+bool SequenceRange::operator>=(const SequenceRange& right) const {
+    return (*this == right) || (*this > right);
+}
+
+bool SequenceRange::operator<=(const SequenceRange& right) const {
+    return (*this == right) || (*this < right);
+}
+
+/// `SequenceRange`s intersects if max delay of one range
```

Comment:
Documentation comments should go in the header, otherwise I don't think they'll be picked up by doxygen into the doc build.